design/pipeline-triggers.md

```diff
@@ -0,0 +1,369 @@
+# Triggers in pipelines
+
+Triggers are events on which you can start your pipeline run automatically. You can enable triggers on your pipeline by subscribing to both internal and external events.
+
```

Comment:
This would be a good place to illustrate with some customer-focused scenarios (just a handful). Then, when we have questions or disagreements about a requirement below, we can use the customer voice as a starting point for resolving.

Reply:
Sure, will add few scenarios.